In [1]:
import pandas as pd
import numpy as np
import requests
import json
from pandas import json_normalize

from geopy.geocoders import Nominatim

import matplotlib
import matplotlib.cm as cm
import matplotlib.colors as colors

from sklearn.cluster import KMeans

import folium

pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

#### Loading Municipal Data

In [2]:
munfile = r'municipal.csv'
mun_on = pd.read_csv(munfile)
mun_on.drop(['Unnamed: 0'],axis=1,inplace=True)

In [3]:
mun_on.head()

Municipality  Population     Area  Density
0      Toronto     2731571   630.20   4334.5
1       Ottawa      934243  2790.30    334.8
2  Mississauga      721599   292.43   2467.6
3     Brampton      593638   266.36   2228.7
4     Hamilton      536917  1117.29    480.6

#### Loading City with Covid Data

In [4]:
covcityfile = r'cov_city.csv'
cov_city = pd.read_csv(covcityfile)
cov_city.drop(['Unnamed: 0'],axis=1,inplace=True)

In [5]:
cov_city_loc = [cov_city['Latitude'].mean(),cov_city['Longitude'].mean()]

In [6]:
map_on = folium.Map(location=[cov_city_loc[0],cov_city_loc[1]], zoom_start=6)

## taken from lab
for lat, lng, mun in zip(cov_city['Latitude'], cov_city['Longitude'], cov_city['Reporting PHU City']):
    label = '{}'.format(mun)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=6,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.5,
        parse_html=False).add_to(map_on) 

map_on

#### Let's look at the Full Covid-19 Positive Cases Data

In [7]:
cov_pos_file = r'conposcovidloc.csv'
cov_pos = pd.read_csv(cov_pos_file)
cov_pos.head()

Row_ID Accurate_Episode_Date Age_Group Client_Gender Case_AcquisitionInfo  \
0       1            2020-01-22       50s        FEMALE       Travel-Related   
1       2            2020-01-21       50s          MALE       Travel-Related   
2       3            2020-01-24       20s        FEMALE       Travel-Related   
3       4            2020-02-05       20s        FEMALE       Travel-Related   
4       5            2020-02-16       60s        FEMALE       Travel-Related   

   Outcome1                 Reporting_PHU           Reporting_PHU_Address  \
0  Resolved         Toronto Public Health  277 Victoria Street, 5th Floor   
1  Resolved         Toronto Public Health  277 Victoria Street, 5th Floor   
2  Resolved  Middlesex-London Health Unit                  50 King Street   
3  Resolved         Toronto Public Health  277 Victoria Street, 5th Floor   
4  Resolved         Toronto Public Health  277 Victoria Street, 5th Floor   

  Reporting_PHU_City Reporting_PHU_Postal_Code  \
0            Toronto                   M5B 1W2   
1            Toronto                   M5B 1W2   
2             London                   N6A 5L7   
3            Toronto                   M5B 1W2   
4            Toronto                   M5B 1W2   

                               Reporting_PHU_Website  Reporting_PHU_Latitude  \
0  www.toronto.ca/community-people/health-wellnes...               43.656591   
1  www.toronto.ca/community-people/health-wellnes...               43.656591   
2                                 www.healthunit.com               42.981468   
3  www.toronto.ca/community-people/health-wellnes...               43.656591   
4  www.toronto.ca/community-people/health-wellnes...               43.656591   

   Reporting_PHU_Longitude  
0               -79.379358  
1               -79.379358  
2               -81.254016  
3               -79.379358  
4               -79.379358

In [8]:
cov_pos_fil = cov_pos[['Row_ID', \
                            'Accurate_Episode_Date', \
                            'Case_AcquisitionInfo' , \
                            'Outcome1', \
                            'Reporting_PHU_City']] \
                            .set_index(['Row_ID'])
cov_pos_fil.columns = ['Date','Infection Source','Outcome','City']

In [9]:
cov_pos_fil.head(10)

Date             Infection Source   Outcome       City
Row_ID                                                              
1       2020-01-22               Travel-Related  Resolved    Toronto
2       2020-01-21               Travel-Related  Resolved    Toronto
3       2020-01-24               Travel-Related  Resolved     London
4       2020-02-05               Travel-Related  Resolved    Toronto
5       2020-02-16               Travel-Related  Resolved    Toronto
6       2020-02-20  Contact of a confirmed case  Resolved    Toronto
7       2020-02-24               Travel-Related  Resolved    Toronto
8       2020-02-25               Travel-Related  Resolved  Newmarket
9       2020-02-20               Travel-Related  Resolved    Toronto
10      2020-02-24               Travel-Related  Resolved     Whitby

#### Let's work on the Foursquare data

In [303]:
CLIENT_ID = 'yourcode' # your Foursquare ID
CLIENT_SECRET = 'yoursecret' # your Foursquare Secret
VERSION = 'yourversion' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: AVKASV1WRO0ILDGAXCZ15WJUWC1BI2OW4OVIWCB1GFFMGHHL
CLIENT_SECRET:NFW422HI05C0QRJLJDGNS0UJ1I2VBH5BILDRXDIZ2B4DRFFU


In [ ]:
CLIENT_ID = 'id' # your Foursquare ID
CLIENT_SECRET = 'secret' # your Foursquare Secret
VERSION = 'version' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET: ' + CLIENT_SECRET)

#### Add the Latitude and Longitude to the Municipality data

#### Same as the Lab

In [11]:
## same as the Lab
# function that extracts the category of the venue

LIMIT=50

def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

In [12]:
def getNearbyVenues(names, latitudes, longitudes, radius=1000):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Municipality', 
                  'Municipality Latitude', 
                  'Municipality Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [13]:
cov_city.head(10)

Reporting PHU City   Latitude  Longitude
0            Toronto  43.653482 -79.383935
1             London  42.983675 -81.249607
2          Newmarket  44.056258 -79.461708
3             Whitby  43.899871 -78.940201
4           Waterloo  43.466874 -80.524635
5        Mississauga  43.590338 -79.645729
6           Oakville  43.447436 -79.666672
7             Ottawa  45.421106 -75.690308
8            Sudbury  46.492720 -80.991211
9           Hamilton  43.256080 -79.872858

In [15]:
cov_city_venues = getNearbyVenues(names=cov_city['Reporting PHU City'],
                                   latitudes=cov_city['Latitude'],
                                   longitudes=cov_city['Longitude']
                                  )

Toronto
London
Newmarket
Whitby
Waterloo
Mississauga
Oakville
Ottawa
Sudbury
Hamilton
Cornwall
Barrie
Thorold
Port Hope
Stratford
Kenora
Peterborough
Owen Sound
Guelph
Brantford
Windsor
Kingston
Sault Ste. Marie
Timmins
Chatham
Belleville
St. Thomas
New Liskeard
North Bay
Point Edward
Thunder Bay
Brockville
Simcoe
Pembroke


In [17]:
print(cov_city_venues.shape)
cov_city_venues.head()

(1049, 7)


Municipality  Municipality Latitude  Municipality Longitude  \
0      Toronto              43.653482              -79.383935   
1      Toronto              43.653482              -79.383935   
2      Toronto              43.653482              -79.383935   
3      Toronto              43.653482              -79.383935   
4      Toronto              43.653482              -79.383935   

                               Venue  Venue Latitude  Venue Longitude  \
0                   Downtown Toronto       43.653232       -79.385296   
1             Nathan Phillips Square       43.652270       -79.383516   
2                             Indigo       43.653515       -79.380696   
3  Eggspectation Bell Trinity Square       43.653144       -79.381980   
4                       Chatime 日出茶太       43.655542       -79.384684   

    Venue Category  
0     Neighborhood  
1            Plaza  
2        Bookstore  
3   Breakfast Spot  
4  Bubble Tea Shop

In [18]:
cov_city_venues.groupby('Municipality').count()

Municipality Latitude  Municipality Longitude  Venue  \
Municipality                                                             
Barrie                               46                      46     46   
Belleville                           28                      28     28   
Brantford                            35                      35     35   
Brockville                           18                      18     18   
Chatham                              21                      21     21   
Cornwall                             21                      21     21   
Guelph                               50                      50     50   
Hamilton                             50                      50     50   
Kenora                               12                      12     12   
Kingston                             50                      50     50   
London                               50                      50     50   
Mississauga                          50                      50     50   
New Liskeard                          8                       8      8   
Newmarket                            32                      32     32   
North Bay                            18                      18     18   
Oakville                             47                      47     47   
Ottawa                               50                      50     50   
Owen Sound                           22                      22     22   
Pembroke                              4                       4      4   
Peterborough                         50                      50     50   
Point Edward                         19                      19     19   
Port Hope                            22                      22     22   
Sault Ste. Marie                     18                      18     18   
Simcoe                                3                       3      3   
St. Thomas                           20                      20     20   
Stratford                            50                      50     50   
Sudbury                              35                      35     35   
Thorold                              13                      13     13   
Thunder Bay                          18                      18     18   
Timmins                              20                      20     20   
Toronto                              50                      50     50   
Waterloo                             50                      50     50   
Whitby                               19                      19     19   
Windsor                              50                      50     50   

                  Venue Latitude  Venue Longitude  Venue Category  
Municipality                                                       
Barrie                        46               46              46  
Belleville                    28               28              28  
Brantford                     35               35              35  
Brockville                    18               18              18  
Chatham                       21               21              21  
Cornwall                      21               21              21  
Guelph                        50               50              50  
Hamilton                      50               50              50  
Kenora                        12               12              12  
Kingston                      50               50              50  
London                        50               50              50  
Mississauga                   50               50              50  
New Liskeard                   8                8               8  
Newmarket                     32               32              32  
North Bay                     18               18              18  
Oakville                      47               47              47  
Ottawa                        50               50              50  
Owen Sound                    22               22            

##### Correcting Lat and Long for Pembroke and Simcoe because the resulting venues are too low

In [19]:
# lat and long by googling manually
pembroke_loc = [45.8267,-77.1109]
simcoe_loc = [42.8373,-80.3040]
#newlisk_loc = [47.5037,-79.6979]

In [20]:
temp_cov = cov_city.set_index(['Reporting PHU City'])
temp_cov.head()
temp_cov.loc[['Pembroke','Simcoe']]
#temp_cov.loc[['Pembroke','Simcoe','New Liskeard']]

Latitude  Longitude
Reporting PHU City                      
Pembroke            45.825540 -77.115065
Simcoe              44.501401 -79.703871

In [21]:
temp_cov.loc[["Pembroke"],["Latitude"]] = pembroke_loc[0]
temp_cov.loc[["Pembroke"],["Longitude"]] = pembroke_loc[1]
#temp_cov.loc[["New Liskeard"],["Latitude"]] = newlisk_loc[0]
#temp_cov.loc[["New Liskeard"],["Longitude"]] = newlisk_loc[1]
temp_cov.loc[["Simcoe"],["Latitude"]] = simcoe_loc[0]
temp_cov.loc[["Simcoe"],["Longitude"]] = simcoe_loc[1]

In [22]:
#temp_cov.loc[['Pembroke','Simcoe','New Liskeard']]
temp_cov.loc[['Pembroke','Simcoe']]

Latitude  Longitude
Reporting PHU City                     
Pembroke             45.8267   -77.1109
Simcoe               42.8373   -80.3040

In [23]:
cov_city = temp_cov.reset_index()
cov_city.tail()

Reporting PHU City   Latitude  Longitude
29       Point Edward  42.997903 -82.413294
30        Thunder Bay  48.406414 -89.259796
31         Brockville  44.589593 -75.684333
32             Simcoe  42.837300 -80.304000
33           Pembroke  45.826700 -77.110900

In [25]:
cov_city_venues = getNearbyVenues(names=cov_city['Reporting PHU City'],
                                   latitudes=cov_city['Latitude'],
                                   longitudes=cov_city['Longitude']
                                  )
print('finished')

Toronto
London
Newmarket
Whitby
Waterloo
Mississauga
Oakville
Ottawa
Sudbury
Hamilton
Cornwall
Barrie
Thorold
Port Hope
Stratford
Kenora
Peterborough
Owen Sound
Guelph
Brantford
Windsor
Kingston
Sault Ste. Marie
Timmins
Chatham
Belleville
St. Thomas
New Liskeard
North Bay
Point Edward
Thunder Bay
Brockville
Simcoe
Pembroke
finished


In [26]:
cov_city_venues.to_csv('cov_city_venues.csv')

In [27]:
print(cov_city_venues.shape)
cov_city_venues.head()

(1060, 7)


Municipality  Municipality Latitude  Municipality Longitude  \
0      Toronto              43.653482              -79.383935   
1      Toronto              43.653482              -79.383935   
2      Toronto              43.653482              -79.383935   
3      Toronto              43.653482              -79.383935   
4      Toronto              43.653482              -79.383935   

                               Venue  Venue Latitude  Venue Longitude  \
0                   Downtown Toronto       43.653232       -79.385296   
1             Nathan Phillips Square       43.652270       -79.383516   
2                             Indigo       43.653515       -79.380696   
3  Eggspectation Bell Trinity Square       43.653144       -79.381980   
4                       Chatime 日出茶太       43.655542       -79.384684   

    Venue Category  
0     Neighborhood  
1            Plaza  
2        Bookstore  
3   Breakfast Spot  
4  Bubble Tea Shop

In [28]:
cov_city_venues.groupby('Municipality').count()

Municipality Latitude  Municipality Longitude  Venue  \
Municipality                                                             
Barrie                               46                      46     46   
Belleville                           28                      28     28   
Brantford                            35                      35     35   
Brockville                           18                      18     18   
Chatham                              21                      21     21   
Cornwall                             21                      21     21   
Guelph                               50                      50     50   
Hamilton                             50                      50     50   
Kenora                               12                      12     12   
Kingston                             50                      50     50   
London                               50                      50     50   
Mississauga                          50                      50     50   
New Liskeard                          8                       8      8   
Newmarket                            32                      32     32   
North Bay                            18                      18     18   
Oakville                             47                      47     47   
Ottawa                               50                      50     50   
Owen Sound                           22                      22     22   
Pembroke                              7                       7      7   
Peterborough                         50                      50     50   
Point Edward                         19                      19     19   
Port Hope                            22                      22     22   
Sault Ste. Marie                     18                      18     18   
Simcoe                               11                      11     11   
St. Thomas                           20                      20     20   
Stratford                            50                      50     50   
Sudbury                              35                      35     35   
Thorold                              13                      13     13   
Thunder Bay                          18                      18     18   
Timmins                              20                      20     20   
Toronto                              50                      50     50   
Waterloo                             50                      50     50   
Whitby                               19                      19     19   
Windsor                              50                      50     50   

                  Venue Latitude  Venue Longitude  Venue Category  
Municipality                                                       
Barrie                        46               46              46  
Belleville                    28               28              28  
Brantford                     35               35              35  
Brockville                    18               18              18  
Chatham                       21               21              21  
Cornwall                      21               21              21  
Guelph                        50               50              50  
Hamilton                      50               50              50  
Kenora                        12               12              12  
Kingston                      50               50              50  
London                        50               50              50  
Mississauga                   50               50              50  
New Liskeard                   8                8               8  
Newmarket                     32               32              32  
North Bay                     18               18              18  
Oakville                      47               47              47  
Ottawa                        50               50              50  
Owen Sound                    22               22            

In [29]:
print('There are {} uniques categories.'.format(len(cov_city_venues['Venue Category'].unique())))

There are 167 uniques categories.


In [30]:
# one hot encoding
cov_city_onehot = pd.get_dummies(cov_city_venues[['Venue Category']], prefix="", prefix_sep="") 
cov_city_onehot.insert(0,'Area',cov_city_venues['Municipality'])
cov_city_onehot.head()

Area  ATM  American Restaurant  Art Gallery  Art Museum  \
0  Toronto    0                    0            0           0   
1  Toronto    0                    0            0           0   
2  Toronto    0                    0            0           0   
3  Toronto    0                    0            0           0   
4  Toronto    0                    0            0           0   

   Arts & Crafts Store  Asian Restaurant  Auto Garage  Automotive Shop  \
0                    0                 0            0                0   
1                    0                 0            0                0   
2                    0                 0            0                0   
3                    0                 0            0                0   
4                    0                 0            0                0   

   BBQ Joint  Bakery  Bank  Bar  Beer Garden  Beer Store  Bistro  Bookstore  \
0          0       0     0    0            0           0       0          0   
1          0       0     0    0            0           0       0          0   
2          0       0     0    0            0           0       0          1   
3          0       0     0    0            0           0       0          0   
4          0       0     0    0            0           0       0          0   

   Border Crossing  Bowling Alley  Breakfast Spot  Brewery  Bubble Tea Shop  \
0                0              0               0        0                0   
1                0              0               0        0                0   
2                0              0               0        0                0   
3                0              0               1        0                0   
4                0              0               0        0                1   

   Burger Joint  Burrito Place  Business Service  Café  \
0             0              0                 0     0   
1             0              0                 0     0   
2             0              0                 0     0   
3             0              0                 0     0   
4             0              0                 0     0   

   Cajun / Creole Restaurant  Canal Lock  Casino  Chinese Restaurant  \
0                          0           0       0                   0   
1                          0           0       0                   0   
2                          0           0       0                   0   
3                          0           0       0                   0   
4                          0           0       0                   0   

   Chocolate Shop  Church  Clothing Store  Cocktail Bar  Coffee Shop  \
0               0       0               0             0            0   
1               0       0               0             0            0   
2               0       0               0             0            0   
3               0       0               0             0            0   
4               0       0               0             0            0   

   College Gym  Comic Shop  Concert Hall  Convenience Store  Cosmetics Shop  \
0            0           0             0                  0               0   
1            0           0             0                  0               0   
2            0           0             0                  0               0   
3            0           0             0                  0               0   
4            0           0             0                  0               0   

   Deli / Bodega  Department Store  Dessert Shop  Diner  Discount Store  \
0              0                 0             0      0               0   
1              0                 0             0      0               0   
2              0                 0             0      0               0   
3              0                 0             0      0               0   
4              0                 0             0      0               0   

   Dog Run  Duty-free Shop  Eastern European Restaurant  Electronics Store  \
0

In [31]:
cov_city_onehot.shape

(1060, 168)

In [32]:
cov_city_grouped = cov_city_onehot.groupby('Area').mean().reset_index()
cov_city_grouped

Area       ATM  American Restaurant  Art Gallery  Art Museum  \
0             Barrie  0.021739             0.021739     0.021739        0.00   
1         Belleville  0.000000             0.000000     0.000000        0.00   
2          Brantford  0.000000             0.000000     0.000000        0.00   
3         Brockville  0.000000             0.000000     0.000000        0.00   
4            Chatham  0.000000             0.047619     0.000000        0.00   
5           Cornwall  0.000000             0.000000     0.000000        0.00   
6             Guelph  0.000000             0.020000     0.000000        0.00   
7           Hamilton  0.000000             0.020000     0.020000        0.00   
8             Kenora  0.000000             0.000000     0.000000        0.00   
9           Kingston  0.000000             0.000000     0.000000        0.00   
10            London  0.000000             0.000000     0.020000        0.00   
11       Mississauga  0.000000             0.000000     0.000000        0.00   
12      New Liskeard  0.000000             0.125000     0.000000        0.00   
13         Newmarket  0.000000             0.000000     0.000000        0.00   
14         North Bay  0.000000             0.000000     0.000000        0.00   
15          Oakville  0.000000             0.000000     0.000000        0.00   
16            Ottawa  0.000000             0.000000     0.040000        0.00   
17        Owen Sound  0.000000             0.045455     0.000000        0.00   
18          Pembroke  0.000000             0.000000     0.000000        0.00   
19      Peterborough  0.000000             0.000000     0.000000        0.00   
20      Point Edward  0.000000             0.052632     0.000000        0.00   
21         Port Hope  0.000000             0.000000     0.000000        0.00   
22  Sault Ste. Marie  0.000000             0.000000     0.000000        0.00   
23            Simcoe  0.000000             0.000000     0.000000        0.00   
24        St. Thomas  0.000000             0.050000     0.000000        0.00   
25         Stratford  0.000000             0.000000     0.000000        0.02   
26           Sudbury  0.000000             0.000000     0.000000        0.00   
27           Thorold  0.000000             0.000000     0.000000        0.00   
28       Thunder Bay  0.000000             0.111111     0.000000        0.00   
29           Timmins  0.000000             0.000000     0.000000        0.00   
30           Toronto  0.000000             0.020000     0.000000        0.02   
31          Waterloo  0.000000             0.000000     0.020000        0.00   
32            Whitby  0.000000             0.000000     0.000000        0.00   
33           Windsor  0.000000             0.000000     0.020000        0.00   

    Arts & Crafts Store  Asian Restaurant  Auto Garage  Automotive Shop  \
0               0.00000          0.000000     0.000000         0.000000   
1               0.00000          0.000000     0.000000         0.035714   
2               0.00000          0.028571     0.000000         0.000000   
3               0.00000          0.000000     0.000000         0.000000   
4               0.00000          0.000000     0.000000         0.000000   
5               0.00000          0.000000     0.047619         0.000000   
6               0.00000          0.000000     0.000000         0.000000   
7               0.00000          0.000000     0.000000         0.000000   
8               0.00000          0.000000     0.000000         0.000000   
9               0.00000          0.000000     0.000000         0.000000   
10              0.00000          0.020000     0.000000         0.000000   
11              0.00000          0.020000     0.000000         0.000000   
12              0.00000          0.000000     0.000000         0.000000   
13              0.03125          0.000000     0.000000         0.000000   
14              0.00000          0.000000     0.000000         0.000000   
15             

In [33]:
num_top_venues = 5

for hood in cov_city_grouped['Area']:
    print("----"+hood+"----")
    #temp = down_grouped[down_grouped['Area'] == hood].T.reset_index()
    temp = cov_city_grouped[cov_city_grouped['Area'] == hood].T.reset_index()
    temp.columns = ['venue','freq']
    temp = temp.iloc[1:]
    temp['freq'] = temp['freq'].astype(float)
    temp = temp.round({'freq': 2})
    print(temp.sort_values('freq', ascending=False).reset_index(drop=True).head(num_top_venues))
    print('\n')

----Barrie----
                           venue  freq
0                    Coffee Shop  0.09
1                            Pub  0.07
2                           Café  0.04
3                 Sandwich Place  0.04
4  Vegetarian / Vegan Restaurant  0.04


----Belleville----
                  venue  freq
0       Harbor / Marina  0.07
1  Gym / Fitness Center  0.04
2     Korean Restaurant  0.04
3           Coffee Shop  0.04
4           Music Store  0.04


----Brantford----
         venue  freq
0  Coffee Shop  0.09
1         Café  0.09
2   Restaurant  0.06
3         Bank  0.06
4  Pizza Place  0.06


----Brockville----
           venue  freq
0       Pharmacy  0.11
1            Pub  0.11
2           Park  0.06
3  Train Station  0.06
4    Coffee Shop  0.06


----Chatham----
               venue  freq
0        Coffee Shop  0.14
1  Convenience Store  0.10
2          Bookstore  0.05
3         Beer Store  0.05
4   Sushi Restaurant  0.05


----Cornwall----
            venue  freq
0   Shopping Mall  0.1

In [34]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

In [35]:
num_top_venues = 5

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Area']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
cov_city_venues_sorted = pd.DataFrame(columns=columns)
cov_city_venues_sorted['Area'] = cov_city_grouped['Area']

for ind in np.arange(cov_city_grouped.shape[0]):
    cov_city_venues_sorted.iloc[ind, 1:] = return_most_common_venues(cov_city_grouped.iloc[ind, :], num_top_venues)

cov_city_venues_sorted.head()

Area 1st Most Common Venue 2nd Most Common Venue  \
0      Barrie           Coffee Shop                   Pub   
1  Belleville       Harbor / Marina           Gas Station   
2   Brantford                  Café           Coffee Shop   
3  Brockville              Pharmacy                   Pub   
4     Chatham           Coffee Shop     Convenience Store   

  3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  
0        Sandwich Place       Harbor / Marina                  Café  
1     French Restaurant          Burger Joint               Library  
2            Restaurant                 Diner  Fast Food Restaurant  
3           Coffee Shop        Sandwich Place     Fish & Chips Shop  
4            Beer Store               Theater            Restaurant

In [36]:
# set number of clusters
kclusters = 4

cov_city_grouped_clustering = cov_city_grouped.drop('Area', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(cov_city_grouped_clustering)

# check cluster labels generated for each row in the dataframe
kmeans.labels_[0:10] 

array([2, 2, 2, 2, 1, 1, 2, 2, 0, 2])

In [37]:
# add clustering labels
cov_city_venues_sorted.insert(0, 'Cluster Labels', kmeans.labels_)

cov_city_merged = cov_city

# merge toronto_grouped with toronto_data to add latitude/longitude for each neighborhood
cov_city_merged = cov_city_merged.join(cov_city_venues_sorted.set_index('Area'), on='Reporting PHU City')

cov_city_merged.head() # check the last columns!

Reporting PHU City   Latitude  Longitude  Cluster Labels  \
0            Toronto  43.653482 -79.383935               2   
1             London  42.983675 -81.249607               2   
2          Newmarket  44.056258 -79.461708               2   
3             Whitby  43.899871 -78.940201               1   
4           Waterloo  43.466874 -80.524635               2   

  1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
0           Coffee Shop                  Café               Theater   
1     Indian Restaurant             Bookstore       Thai Restaurant   
2     Electronics Store                  Café          Burger Joint   
3           Coffee Shop                  Bank        Sandwich Place   
4            Restaurant                  Café                   Bar   

  4th Most Common Venue 5th Most Common Venue  
0            Restaurant                 Plaza  
1    Italian Restaurant                 Diner  
2                Bakery             Gastropub  
3           Pizza Place              Pharmacy  
4           Coffee Shop           Pizza Place

In [38]:
cov_city_merged.tail()

Reporting PHU City   Latitude  Longitude  Cluster Labels  \
29       Point Edward  42.997903 -82.413294               2   
30        Thunder Bay  48.406414 -89.259796               1   
31         Brockville  44.589593 -75.684333               2   
32             Simcoe  42.837300 -80.304000               0   
33           Pembroke  45.826700 -77.110900               3   

   1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
29            Food Truck        Ice Cream Shop            Restaurant   
30            Restaurant           Coffee Shop   American Restaurant   
31              Pharmacy                   Pub           Coffee Shop   
32  Fast Food Restaurant           Golf Course        Sandwich Place   
33        Ice Cream Shop           Supermarket     Electronics Store   

    4th Most Common Venue 5th Most Common Venue  
29                  Hotel       Border Crossing  
30                  Hotel         Grocery Store  
31         Sandwich Place     Fish & Chips Shop  
32  Portuguese Restaurant           Supermarket  
33            Coffee Shop                  Café

In [39]:
# create map
map_on_clusters = folium.Map(location=[cov_city_merged['Latitude'].mean(),cov_city_merged['Longitude'].mean()], zoom_start=6)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(cov_city_merged['Latitude'], cov_city_merged['Longitude'], cov_city_merged['Reporting PHU City'], cov_city_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=10,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_on_clusters)
       
map_on_clusters

In [40]:
cov_city_merged.loc[cov_city_merged['Cluster Labels'] == 0, cov_city_merged.columns[[0] + list(range(3, cov_city_merged.shape[1]))]]

Reporting PHU City  Cluster Labels 1st Most Common Venue  \
15             Kenora               0  Fast Food Restaurant   
23            Timmins               0  Fast Food Restaurant   
32             Simcoe               0  Fast Food Restaurant   

   2nd Most Common Venue 3rd Most Common Venue  4th Most Common Venue  \
15              Pharmacy           Supermarket        Harbor / Marina   
23           Coffee Shop           Supermarket            Gas Station   
32           Golf Course        Sandwich Place  Portuguese Restaurant   

   5th Most Common Venue  
15         Grocery Store  
23              Pharmacy  
32           Supermarket

In [41]:
cov_city_merged.loc[cov_city_merged['Cluster Labels'] == 1, cov_city_merged.columns[[0] + list(range(3, cov_city_merged.shape[1]))]]

Reporting PHU City  Cluster Labels 1st Most Common Venue  \
3              Whitby               1           Coffee Shop   
10           Cornwall               1         Shopping Mall   
13          Port Hope               1           Coffee Shop   
24            Chatham               1           Coffee Shop   
26         St. Thomas               1           Coffee Shop   
27       New Liskeard               1         Grocery Store   
28          North Bay               1            Restaurant   
30        Thunder Bay               1            Restaurant   

   2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
3                   Bank        Sandwich Place           Pizza Place   
10          Burger Joint           Coffee Shop              Pharmacy   
13    Italian Restaurant             Pet Store                  Bank   
24     Convenience Store            Beer Store               Theater   
26     Convenience Store           Pizza Place                Museum   
27            Beer Store   American Restaurant                 Hotel   
28                   Gym           Coffee Shop        Sandwich Place   
30           Coffee Shop   American Restaurant                 Hotel   

   5th Most Common Venue  
3               Pharmacy  
10                Bakery  
13                  Park  
24            Restaurant  
26              Pharmacy  
27        Hardware Store  
28           Pizza Place  
30         Grocery Store

In [42]:
cov_city_merged.loc[cov_city_merged['Cluster Labels'] == 2, cov_city_merged.columns[[0] + list(range(3, cov_city_merged.shape[1]))]]

Reporting PHU City  Cluster Labels 1st Most Common Venue  \
0             Toronto               2           Coffee Shop   
1              London               2     Indian Restaurant   
2           Newmarket               2     Electronics Store   
4            Waterloo               2            Restaurant   
5         Mississauga               2           Coffee Shop   
6            Oakville               2           Coffee Shop   
7              Ottawa               2           Coffee Shop   
8             Sudbury               2                 Hotel   
9            Hamilton               2                  Café   
11             Barrie               2           Coffee Shop   
12            Thorold               2                Tunnel   
14          Stratford               2                 Hotel   
16       Peterborough               2                  Café   
17         Owen Sound               2                  Café   
18             Guelph               2                  Café   
19          Brantford               2                  Café   
20            Windsor               2                  Café   
21           Kingston               2                   Pub   
22   Sault Ste. Marie               2      Department Store   
25         Belleville               2       Harbor / Marina   
29       Point Edward               2            Food Truck   
31         Brockville               2              Pharmacy   

   2nd Most Common Venue 3rd Most Common Venue      4th Most Common Venue  \
0                   Café               Theater                 Restaurant   
1              Bookstore       Thai Restaurant         Italian Restaurant   
2                   Café          Burger Joint                     Bakery   
4                   Café                   Bar                Coffee Shop   
5         Clothing Store             Bookstore             Cosmetics Shop   
6             Restaurant                   Pub         Italian Restaurant   
7           Concert Hall                 Hotel              Grocery Store   
8                   Café                  Bank                Coffee Shop   
9                    Pub            Restaurant  Middle Eastern Restaurant   
11                   Pub        Sandwich Place            Harbor / Marina   
12                   Bar                Bakery                  Gastropub   
14            Restaurant             Gastropub             History Museum   
16    Mexican Restaurant            Restaurant                        Pub   
17                 Hotel              Pharmacy                 Beer Store   
18                   Pub           Coffee Shop          Indian Restaurant   
19           Coffee Shop            Restaurant                      Diner   
20                   Bar            Restaurant                        Pub   
21                  Park                 Hotel         Italian Restaurant   
22                  Bank         Shopping Mall               Burger Joint   
25           Gas Station     French Restaurant               Burger Joint   
29        Ice Cream Shop            Restaurant                      Hotel   
31                   Pub           Coffee Shop             Sandwich Place   

    5th Most Common Venue  
0                   Plaza  
1                   Diner  
2               Gastropub  
4             Pizza Place  
5      Italian Restaurant  
6                  Bakery  
7             Art Gallery  
8          Sandwich Place  
9   Vietnamese Restaurant  
11                   Café  
12         Sandwich Place  
14     Italian Restaurant  
16            Coffee Shop  
17            Pizza Place  
18                  Diner  
19   Fast Food Restaurant  
20           Burger Joint  
21                   Café  
22                   Food  
25                Library  
29        Border Crossing  
31      Fish & Chips Shop

In [43]:
cov_city_merged.loc[cov_city_merged['Cluster Labels'] == 3, cov_city_merged.columns[[0] + list(range(3, cov_city_merged.shape[1]))]]

Reporting PHU City  Cluster Labels 1st Most Common Venue  \
33           Pembroke               3        Ice Cream Shop   

   2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
33           Supermarket     Electronics Store           Coffee Shop   

   5th Most Common Venue  
33                  Café

In [44]:
#cov_city_merged.loc[cov_city_merged['Cluster Labels'] == 4, cov_city_merged.columns[[0] + list(range(3, cov_city_merged.shape[1]))]]

#### Checking the positive cases data

In [45]:
cov_pos_group = cov_pos_fil.groupby(['Infection Source']).count()
cov_pos_group.head()

Date  Outcome  City
Infection Source                                
Contact of a confirmed case  2205     2205  2205
Information pending          4929     4929  4929
Neither                      3836     3836  3836
Travel-Related               1275     1275  1275

In [46]:
cov_city_merged.head()

Reporting PHU City   Latitude  Longitude  Cluster Labels  \
0            Toronto  43.653482 -79.383935               2   
1             London  42.983675 -81.249607               2   
2          Newmarket  44.056258 -79.461708               2   
3             Whitby  43.899871 -78.940201               1   
4           Waterloo  43.466874 -80.524635               2   

  1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
0           Coffee Shop                  Café               Theater   
1     Indian Restaurant             Bookstore       Thai Restaurant   
2     Electronics Store                  Café          Burger Joint   
3           Coffee Shop                  Bank        Sandwich Place   
4            Restaurant                  Café                   Bar   

  4th Most Common Venue 5th Most Common Venue  
0            Restaurant                 Plaza  
1    Italian Restaurant                 Diner  
2                Bakery             Gastropub  
3           Pizza Place              Pharmacy  
4           Coffee Shop           Pizza Place

In [47]:
cov_city_merged.rename(columns={'Reporting PHU City' : 'City'}).head()

City   Latitude  Longitude  Cluster Labels 1st Most Common Venue  \
0    Toronto  43.653482 -79.383935               2           Coffee Shop   
1     London  42.983675 -81.249607               2     Indian Restaurant   
2  Newmarket  44.056258 -79.461708               2     Electronics Store   
3     Whitby  43.899871 -78.940201               1           Coffee Shop   
4   Waterloo  43.466874 -80.524635               2            Restaurant   

  2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
0                  Café               Theater            Restaurant   
1             Bookstore       Thai Restaurant    Italian Restaurant   
2                  Café          Burger Joint                Bakery   
3                  Bank        Sandwich Place           Pizza Place   
4                  Café                   Bar           Coffee Shop   

  5th Most Common Venue  
0                 Plaza  
1                 Diner  
2             Gastropub  
3              Pharmacy  
4           Pizza Place

In [48]:
df_cov = cov_city_merged.rename(columns={'Reporting PHU City' : 'City'})

In [49]:
#df_cov = df_cov[['City','Cluster Labels','Latitude','Longitude']]
df_cov.head()

City   Latitude  Longitude  Cluster Labels 1st Most Common Venue  \
0    Toronto  43.653482 -79.383935               2           Coffee Shop   
1     London  42.983675 -81.249607               2     Indian Restaurant   
2  Newmarket  44.056258 -79.461708               2     Electronics Store   
3     Whitby  43.899871 -78.940201               1           Coffee Shop   
4   Waterloo  43.466874 -80.524635               2            Restaurant   

  2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
0                  Café               Theater            Restaurant   
1             Bookstore       Thai Restaurant    Italian Restaurant   
2                  Café          Burger Joint                Bakery   
3                  Bank        Sandwich Place           Pizza Place   
4                  Café                   Bar           Coffee Shop   

  5th Most Common Venue  
0                 Plaza  
1                 Diner  
2             Gastropub  
3              Pharmacy  
4           Pizza Place

In [50]:
cov_pos_fil.head()

Date Infection Source   Outcome     City
Row_ID                                                
1       2020-01-22   Travel-Related  Resolved  Toronto
2       2020-01-21   Travel-Related  Resolved  Toronto
3       2020-01-24   Travel-Related  Resolved   London
4       2020-02-05   Travel-Related  Resolved  Toronto
5       2020-02-16   Travel-Related  Resolved  Toronto

In [51]:
cov_pos_fil.groupby(['City']).count()

Date  Infection Source  Outcome
City                                             
Barrie             209               209      209
Belleville          35                35       35
Brantford           82                82       82
Brockville         279               279      279
Chatham             31                31       31
Cornwall            70                70       70
Guelph             199               199      199
Hamilton           342               342      342
Kenora              13                13       13
Kingston            58                58       58
London             349               349      349
Mississauga       1805              1805     1805
New Liskeard        12                12       12
Newmarket         1157              1157     1157
North Bay           14                14       14
Oakville           393               393      393
Ottawa             913               913      913
Owen Sound          63                63       63
Pembroke            14                14       14
Peterborough        59                59       59
Point Edward       146               146      146
Port Hope          137               137      137
Sault Ste. Marie    12                12       12
Simcoe             164               164      164
St. Thomas          48                48       48
Stratford           36                36       36
Sudbury             46                46       46
Thorold            388               388      388
Thunder Bay         51                51       51
Timmins             52                52       52
Toronto           3584              3584     3584
Waterloo           283               283      283
Whitby             713               713      713
Windsor            488               488      488

In [52]:
df_cov_merge = df_cov

In [53]:
df_cov_merge

City   Latitude  Longitude  Cluster Labels  \
0            Toronto  43.653482 -79.383935               2   
1             London  42.983675 -81.249607               2   
2          Newmarket  44.056258 -79.461708               2   
3             Whitby  43.899871 -78.940201               1   
4           Waterloo  43.466874 -80.524635               2   
5        Mississauga  43.590338 -79.645729               2   
6           Oakville  43.447436 -79.666672               2   
7             Ottawa  45.421106 -75.690308               2   
8            Sudbury  46.492720 -80.991211               2   
9           Hamilton  43.256080 -79.872858               2   
10          Cornwall  45.018442 -74.728703               1   
11            Barrie  44.389311 -79.690174               2   
12           Thorold  43.117935 -79.202235               2   
13         Port Hope  43.951575 -78.293970               1   
14         Stratford  52.192788  -1.706367               2   
15            Kenora  49.766215 -94.489353               0   
16      Peterborough  44.304801 -78.319950               2   
17        Owen Sound  44.566809 -80.944125               2   
18            Guelph  43.546052 -80.249328               2   
19         Brantford  43.140816 -80.263173               2   
20           Windsor  42.317099 -83.035343               2   
21          Kingston  44.230687 -76.481323               2   
22  Sault Ste. Marie  46.523910 -84.320068               2   
23           Timmins  48.477473 -81.330414               0   
24           Chatham  42.405722 -82.185384               1   
25        Belleville  44.163326 -77.385256               2   
26        St. Thomas  42.779022 -81.192988               1   
27      New Liskeard  47.508731 -79.678903               1   
28         North Bay  46.309211 -79.460762               1   
29      Point Edward  42.997903 -82.413294               2   
30       Thunder Bay  48.406414 -89.259796               1   
31        Brockville  44.589593 -75.684333               2   
32            Simcoe  42.837300 -80.304000               0   
33          Pembroke  45.826700 -77.110900               3   

   1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
0            Coffee Shop                  Café               Theater   
1      Indian Restaurant             Bookstore       Thai Restaurant   
2      Electronics Store                  Café          Burger Joint   
3            Coffee Shop                  Bank        Sandwich Place   
4             Restaurant                  Café                   Bar   
5            Coffee Shop        Clothing Store             Bookstore   
6            Coffee Shop            Restaurant                   Pub   
7            Coffee Shop          Concert Hall                 Hotel   
8                  Hotel                  Café                  Bank   
9                   Café                   Pub            Restaurant   
10         Shopping Mall          Burger Joint           Coffee Shop   
11           Coffee Shop                   Pub        Sandwich Place   
12                Tunnel                   Bar                Bakery   
13           Coffee Shop    Italian Restaurant             Pet Store   
14                 Hotel            Restaurant             Gastropub   
15  Fast Food Restaurant              Pharmacy           Supermarket   
16                  Café    Mexican Restaurant            Restaurant   
17                  Café                 Hotel              Pharmacy   
18                  Café                   Pub           Coffee Shop   
19                  Café           Coffee Shop            Restaurant   
20                  Café                   Bar            Restaurant   
21                   Pub                  Park                 Hotel   
22      Department Store                  Bank         Shopping Mall   
23  Fast Food Restaurant           Coffee Shop           Supermarket   
24           Coffee Shop     Convenience Stor

In [54]:
sContact = []
sTravel = []
sPending = []
sNeither = []

def loadCaseTable(city):
    temp = cov_pos_fil[cov_pos_fil['City']==city]
    temp2 = temp.groupby(['Infection Source']).count().T.iloc[[0]]
    temp2.columns.name = ''
    temp2.reset_index(drop=True)
    if(len(temp2.iloc[0]) == 3):
        sContact.append(temp2.iloc[0][0])
        sTravel.append(temp2.iloc[0][2])
        sPending.append(0)
        sNeither.append(temp2.iloc[0][1])
        #print(temp2.columns)
    else:
        sContact.append(temp2.iloc[0][0])
        sTravel.append(temp2.iloc[0][3])
        sPending.append(temp2.iloc[0][1])
        sNeither.append(temp2.iloc[0][2])
        #print('none')

for city in df_cov_merge['City']:
    loadCaseTable(city)
    #print(city)

In [55]:
df_cov_merge['Contact']=sContact
df_cov_merge['Travel']=sTravel
df_cov_merge['Pending']=sPending
df_cov_merge['Neither']=sNeither
df_cov_merge

City   Latitude  Longitude  Cluster Labels  \
0            Toronto  43.653482 -79.383935               2   
1             London  42.983675 -81.249607               2   
2          Newmarket  44.056258 -79.461708               2   
3             Whitby  43.899871 -78.940201               1   
4           Waterloo  43.466874 -80.524635               2   
5        Mississauga  43.590338 -79.645729               2   
6           Oakville  43.447436 -79.666672               2   
7             Ottawa  45.421106 -75.690308               2   
8            Sudbury  46.492720 -80.991211               2   
9           Hamilton  43.256080 -79.872858               2   
10          Cornwall  45.018442 -74.728703               1   
11            Barrie  44.389311 -79.690174               2   
12           Thorold  43.117935 -79.202235               2   
13         Port Hope  43.951575 -78.293970               1   
14         Stratford  52.192788  -1.706367               2   
15            Kenora  49.766215 -94.489353               0   
16      Peterborough  44.304801 -78.319950               2   
17        Owen Sound  44.566809 -80.944125               2   
18            Guelph  43.546052 -80.249328               2   
19         Brantford  43.140816 -80.263173               2   
20           Windsor  42.317099 -83.035343               2   
21          Kingston  44.230687 -76.481323               2   
22  Sault Ste. Marie  46.523910 -84.320068               2   
23           Timmins  48.477473 -81.330414               0   
24           Chatham  42.405722 -82.185384               1   
25        Belleville  44.163326 -77.385256               2   
26        St. Thomas  42.779022 -81.192988               1   
27      New Liskeard  47.508731 -79.678903               1   
28         North Bay  46.309211 -79.460762               1   
29      Point Edward  42.997903 -82.413294               2   
30       Thunder Bay  48.406414 -89.259796               1   
31        Brockville  44.589593 -75.684333               2   
32            Simcoe  42.837300 -80.304000               0   
33          Pembroke  45.826700 -77.110900               3   

   1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
0            Coffee Shop                  Café               Theater   
1      Indian Restaurant             Bookstore       Thai Restaurant   
2      Electronics Store                  Café          Burger Joint   
3            Coffee Shop                  Bank        Sandwich Place   
4             Restaurant                  Café                   Bar   
5            Coffee Shop        Clothing Store             Bookstore   
6            Coffee Shop            Restaurant                   Pub   
7            Coffee Shop          Concert Hall                 Hotel   
8                  Hotel                  Café                  Bank   
9                   Café                   Pub            Restaurant   
10         Shopping Mall          Burger Joint           Coffee Shop   
11           Coffee Shop                   Pub        Sandwich Place   
12                Tunnel                   Bar                Bakery   
13           Coffee Shop    Italian Restaurant             Pet Store   
14                 Hotel            Restaurant             Gastropub   
15  Fast Food Restaurant              Pharmacy           Supermarket   
16                  Café    Mexican Restaurant            Restaurant   
17                  Café                 Hotel              Pharmacy   
18                  Café                   Pub           Coffee Shop   
19                  Café           Coffee Shop            Restaurant   
20                  Café                   Bar            Restaurant   
21                   Pub                  Park                 Hotel   
22      Department Store                  Bank         Shopping Mall   
23  Fast Food Restaurant           Coffee Shop           Supermarket   
24           Coffee Shop     Convenience Stor

In [56]:
df_cov_merge['Total'] = df_cov_merge['Contact'] + df_cov_merge['Travel'] + df_cov_merge['Pending'] + df_cov_merge['Neither']
df_cov_merge.head()

City   Latitude  Longitude  Cluster Labels 1st Most Common Venue  \
0    Toronto  43.653482 -79.383935               2           Coffee Shop   
1     London  42.983675 -81.249607               2     Indian Restaurant   
2  Newmarket  44.056258 -79.461708               2     Electronics Store   
3     Whitby  43.899871 -78.940201               1           Coffee Shop   
4   Waterloo  43.466874 -80.524635               2            Restaurant   

  2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
0                  Café               Theater            Restaurant   
1             Bookstore       Thai Restaurant    Italian Restaurant   
2                  Café          Burger Joint                Bakery   
3                  Bank        Sandwich Place           Pizza Place   
4                  Café                   Bar           Coffee Shop   

  5th Most Common Venue  Contact  Travel  Pending  Neither  Total  
0                 Plaza      424     186     2032      942   3584  
1                 Diner        9      22      312        6    349  
2             Gastropub      221     135      360      441   1157  
3              Pharmacy      152      42       37      482    713  
4           Pizza Place       75      39       78       91    283

In [57]:
df_cov_merge.groupby(['Cluster Labels']).sum()

Latitude    Longitude  Contact  Travel  Pending  Neither  \
Cluster Labels                                                               
0               141.080988  -256.123767       60      21       33      115   
1               360.278989  -643.740707      261     131       56      628   
2               976.448999 -1676.181632     1882    1119     4834     3091   
3                45.826700   -77.110900        2       4        6        2   

                Total  
Cluster Labels         
0                 229  
1                1076  
2               10926  
3                  14

In [58]:
cluster_0 = df_cov_merge.loc[df_cov_merge['Cluster Labels'] == 0, df_cov_merge.columns[[0] + [4] + list(range(9, df_cov_merge.shape[1]))]]
cluster_0.reset_index(drop=True)

City 1st Most Common Venue  Contact  Travel  Pending  Neither  Total
0   Kenora  Fast Food Restaurant        1       9        0        3     13
1  Timmins  Fast Food Restaurant       25       6        4       17     52
2   Simcoe  Fast Food Restaurant       34       6       29       95    164

In [59]:
print('Average of total case per City in Cluster 0: %.2f' % cluster_0['Total'].mean())
print('Average of case per City in Cluster 0 caused by contact with confirmed case or unknown cases: %.2f' % (cluster_0['Contact']+cluster_0['Neither']).mean())

Average of total case per City in Cluster 0: 76.33
Average of case per City in Cluster 0 caused by contact with confirmed case or unknown cases: 58.33


In [60]:
cluster_1 = df_cov_merge.loc[df_cov_merge['Cluster Labels'] == 1, df_cov_merge.columns[[0] + [4] + list(range(9, df_cov_merge.shape[1]))]]
cluster_1.reset_index(drop=True)

City 1st Most Common Venue  Contact  Travel  Pending  Neither  \
0        Whitby           Coffee Shop      152      42       37      482   
1      Cornwall         Shopping Mall       15      30        1       24   
2     Port Hope           Coffee Shop       27      25        7       78   
3       Chatham           Coffee Shop       13       5        1       12   
4    St. Thomas           Coffee Shop       26      11        3        8   
5  New Liskeard         Grocery Store        6       2        0        4   
6     North Bay            Restaurant        4       7        0        3   
7   Thunder Bay            Restaurant       18       9        7       17   

   Total  
0    713  
1     70  
2    137  
3     31  
4     48  
5     12  
6     14  
7     51

In [61]:
print('Average of total case per City in Cluster 1: %.2f' % cluster_1['Total'].mean())
print('Average of case per City in Cluster 1 caused by contact with confirmed case or unknown cases: %.2f' % (cluster_1['Contact']+cluster_1['Neither']).mean())

Average of total case per City in Cluster 1: 134.50
Average of case per City in Cluster 1 caused by contact with confirmed case or unknown cases: 111.12


In [62]:
cluster_2 = df_cov_merge.loc[df_cov_merge['Cluster Labels'] == 2, df_cov_merge.columns[[0] + [4] + list(range(9, df_cov_merge.shape[1]))]]
cluster_2.reset_index(drop=True)

City 1st Most Common Venue  Contact  Travel  Pending  Neither  \
0            Toronto           Coffee Shop      424     186     2032      942   
1             London     Indian Restaurant        9      22      312        6   
2          Newmarket     Electronics Store      221     135      360      441   
3           Waterloo            Restaurant       75      39       78       91   
4        Mississauga           Coffee Shop       71      65     1339      330   
5           Oakville           Coffee Shop      102      76       38      177   
6             Ottawa           Coffee Shop      215     161      143      394   
7            Sudbury                 Hotel       12      20        5        9   
8           Hamilton                  Café      158      58       22      104   
9             Barrie           Coffee Shop       59      46        6       98   
10           Thorold                Tunnel       96      73       13      206   
11         Stratford                 Hotel       16       6        7        7   
12      Peterborough                  Café       17      28        4       10   
13        Owen Sound                  Café       14       9       14       26   
14            Guelph                  Café       70      17       66       46   
15         Brantford                  Café       34      18        8       22   
16           Windsor                  Café       49      90      281       68   
17          Kingston                   Pub       27      19        3        9   
18  Sault Ste. Marie      Department Store        2       7        0        3   
19        Belleville       Harbor / Marina        9      14        7        5   
20      Point Edward            Food Truck       66       7       15       58   
21        Brockville              Pharmacy      136      23       81       39   

    Total  
0    3584  
1     349  
2    1157  
3     283  
4    1805  
5     393  
6     913  
7      46  
8     342  
9     209  
10    388  
11     36  
12     59  
13     63  
14    199  
15     82  
16    488  
17     58  
18     12  
19     35  
20    146  
21    279

In [63]:
print('Average of total case per City in Cluster 0: %.2f' % cluster_2['Total'].mean())
print('Average of case per City in Cluster 2 caused by contact with confirmed case or unknown cases: %.2f' % (cluster_2['Contact']+cluster_2['Neither']).mean())

Average of total case per City in Cluster 0: 496.64
Average of case per City in Cluster 2 caused by contact with confirmed case or unknown cases: 226.05


In [64]:
cluster_3 = df_cov_merge.loc[df_cov_merge['Cluster Labels'] == 3, df_cov_merge.columns[[0] + [4] + list(range(9, df_cov_merge.shape[1]))]]
cluster_3.reset_index(drop=True)

City 1st Most Common Venue  Contact  Travel  Pending  Neither  Total
0  Pembroke        Ice Cream Shop        2       4        6        2     14

In [65]:
print('Average of total case per City in Cluster 0: %.2f' % cluster_3['Total'].mean())
print('Average of case per City in Cluster 3 caused by contact with confirmed case or unknown cases: %.2f' % (cluster_3['Contact']+cluster_3['Neither']).mean())

Average of total case per City in Cluster 0: 14.00
Average of case per City in Cluster 3 caused by contact with confirmed case or unknown cases: 4.00


##### Resulting Table

| Cluster | Num of City/Municipal | Average Total | Average Contact or Neither  |
|:-------:|:---------------------:|:-------------:|:----------------------------|
| 0       | 3   | 76.33   | 58.33  |
| 1       | 8  | 134.50  | 111.12 |
| 2       | 22  | 496.64  | 226.05 |
| 3       | 1   | 14.00   | 4.00   |
